src/Lecture4.hs

```diff
@@ -197,8 +238,33 @@ The 'Stats' data type has multiple fields. All these fields have
 instance for the 'Stats' type itself.
 -}
 
-instance Semigroup Stats where
+evaluate :: Semigroup a => Maybe a -> Maybe a -> Maybe a
+evaluate x y = evaluated
+    where
+        accumulated = x <> y
+        evaluated = case accumulated of
+            Nothing -> Nothing
+            Just !z -> Just z
 
+instance Semigroup Stats where
+    (<>) :: Stats -> Stats -> Stats
+    (Stats totalPos1 totalSum1 absMax1 absMin1 sellMax1 sellMin1 buyMax1 buyMin1 longest1)
+        <> (Stats totalPos2 totalSum2 absMax2 absMin2 sellMax2 sellMin2 buyMax2 buyMin2 longest2) =
+            evaluatedStats where
+
+            !evaluatedSellMax = evaluate sellMax1 sellMax2
```

Comment:
I think the '!' in front of the 'evaluated' variables is not necessary anymore?
The 'evaluate' function might be sufficient, something to test.

Reply:
Yep, I think it should be pretty safe to delete `!` here 🧹